storage/core/src/main/java/com/hortonworks/registries/storage/TransactionalStorageManager.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2017 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.registries.storage;
+
+import com.hortonworks.registries.storage.exception.StorageException;
+
+public interface TransactionalStorageManager extends StorageManager {
```

Comment:
We can have this interface as `TransactionManager` or with some other appropriate name and it does not extend `StorageManager`. So, this interface defines only about transactional behavior. 
 Any `StorageManager` which implements transactional behavior can implement this interface. `TransactionEventListener` can be instantiated with respective storagemanager or noop TransactionManager.